### PR TITLE
Superbill and CMS 1500 Bots

### PR DESCRIPTION
**Pull request by ksmith94**
Related to #4556 this will create bots related to two common billing activities.
Superbill: This takes an Encounter resource and creates a pdf of a superbill, showing all rendered services and related fees.
CMS 1500: This takes a Claim resource and fills out the CMS 1500 form, returning it as a text document. Testing is still a work in progress for this bot

### CODE REVIEW

File: examples/medplum-demo-bots/src/billing-bots/cms-1500.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+  ];
+
+  let cms1500Text: string = '';
+
+  for (const line of cms1500Lines) {
+    cms1500Text += line + '\n';
+  }
+
+  console.log(cms1500Text);
+  return cms1500Text;
```
Comment (reshmakh) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500.ts:157]: We should save this as Document (or something) and link to patient and Encounter right?  @rahul1 - what's the right spot?

File: examples/medplum-demo-bots/src/billing-bots/cms-1500.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+describe('CMS 1500 tests', async () => {
+  beforeAll(() => {
+    indexStructureDefinitionBundle(readJson('fhir/r4/profiles-types.json') as Bundle);
+    indexStructureDefinitionBundle(readJson('fhir/r4/profiles-resources.json') as Bundle);
+    for (const filename of SEARCH_PARAMETER_BUNDLE_FILES) {
+      indexSearchParameterBundle(readJson(filename) as Bundle<SearchParameter>);
+    }
+  });
+
+  test('Fullly answered CMS1500', async () => {
```
Comment (reshmakh) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500.test.ts:19]: Typo

File: examples/medplum-demo-bots/src/billing-bots/cms-1500.test.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    const patientReferralInfo = getReferralInfo({
+      resourceType: 'Patient',
+    });
+    const organizationReferralInfo = getReferralInfo({
+      resourceType: 'Organization',
+      name: 'Referrer',
+      identifier: [
+        {
+          type: {
+            coding: [{ code: 'NPI' }],
```
Comment (reshmakh) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500.test.ts:71]: Does this need to be somethign like https://hl7.org/fhir/sid/us-npi ?  as we show in other places in our documentation?

File: examples/medplum-demo-bots/src/billing-bots/cms-1500.ts [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+  Device,
+  Organization,
+  Patient,
+  Practitioner,
+  PractitionerRole,
+  RelatedPerson,
+} from '@medplum/fhirtypes';
+
+export async function handler(medplum: MedplumClient, event: BotEvent<Claim>): Promise<string> {
+  const claim = event.input;
```
Comment (reshmakh) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500.ts:23]: Maybe creating a DocumentReference with Media and linking to Claim might be the right call here.  Not sure what your perspective is @rahul1

File: examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+            productOrService: {
+              coding: [{ system: 'http://hl7.org/fhir/ValueSet/service-uscls', code: '1201', display: 'Exam, recall' }],
+            },
+            locationAddress: {
+              line: ['289 Johnson Street'],
+              city: 'Ames',
+              state: 'IA',
+            },
+            category: {
+              coding: [{ system: 'http://example.org/claim-item-category', code: 'EMG', display: 'Emergency' }],
```
Comment (ksmith94) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts:326]: I couldn't find a code for an emergency cateogry

File: examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+            sequence: 4,
+            timingPeriod: {
+              start: '2024-03-30',
+              end: '2024-05-21',
+            },
+          },
+          {
+            category: {
+              coding: [
+                { system: 'http://terminology.hl7.org/CodeSystem/claiminformationcategory', code: 'patientpaid' },
```
Comment (ksmith94) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts:260]: This is the system that is used as an example in FHIR, but the `patientpaid` code isn't actually in it

File: examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+              end: '2024-04-20',
+            },
+          },
+          {
+            category: {
+              coding: [{ code: 'info', system: 'http://terminology.hl7.org/CodeSystem/claiminformationcategory' }],
+            },
+            sequence: 2,
+            code: {
+              coding: [{ system: 'http://example.org/info-codes', code: 'patientaccount' }],
```
Comment (ksmith94) [inline examples/medplum-demo-bots/src/billing-bots/cms-1500-test-data.ts:234]: I couldn't find a valid coding for this in FHIR

**Comment by ksmith94** [comment]
@rahul1 I added comments to the code systems that I couldn't find the proper codes for.

**Comment by reshmakh** [comment]
Suggestion here.  If we can't find ValueSets for all of these, can we make some that represent what's needed and add them as a bundle?  This example is an FAQ.

**Review by rahul1 — APPROVED** [review]

**Pull request merged**